Set Up TorchRec Docs

PaulZhang12 opened the pull request:
Differential Revision: D63548309

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+TorchRec is routinely tested on AWS Linux only and should work in similar environments.
+Below demonstrates the compatability matrix that is currently tested:
+
+.. list-table:: TorchRec Compatability
```
It looks like it adds the [title after the table](https://docs-preview.pytorch.org/pytorch/torchrec/2434/setup-torchrec.html#system-requirements) for some reason so maybe just remove the title.
```suggestion
.. list-table::
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+
+TorchRec and FBGEMM have matching version numbers that are tested together upon release:
+
+* TorchRec 1.0 is compatible with FBGEMM 1.0
```
maybe make those two tables as well?

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+Run TorchRec Code
+-----------------
+Now that we have TorchRec properly set up, let's run some TorchRec code!
+Below, we'll run a simple forward pass with TorchRec data types: ``KeyedJaggedTensor`` and ``EmbeddingBagCollection``
```
```suggestion
Below, we'll run a simple forward pass with TorchRec data types: ``KeyedJaggedTensor`` and ``EmbeddingBagCollection``:
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+* TorchRec 0.8 is compatible with PyTorch 2.4
+* TorchRec 0.8 may not be compatible with PyTorch 2.3
+
+Installations
```
```suggestion
Installation
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+
+==================
```
```suggestion
====================
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+
+==================
+Setting up TorchRec
+==================
```
```suggestion
=====================
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+You also have the ability to build TorchRec from source to develop with the latest
+changes in TorchRec. To build from source, check out this `reference <https://github.com/pytorch/torchrec?tab=readme-ov-file#from-source>`_.
+
+Run TorchRec Code
```
```suggestion
Run a Simple TorchRec Example
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+changes in TorchRec. To build from source, check out this `reference <https://github.com/pytorch/torchrec?tab=readme-ov-file#from-source>`_.
+
+Run TorchRec Code
+-----------------
```
```suggestion
------------------------------
```

svekars commented on docs/source/stable/datatypes-api-reference.rst:
```diff
@@ -0,0 +1,35 @@
+There are 3 main classes:
+
+* **JaggedTensor**:
+    A JaggedTensor is a tensor with a jagged dimension which is dimension
```
```suggestion
    A JaggedTensor is a tensor with a jagged dimension meaning
```

svekars commented on docs/source/stable/datatypes-api-reference.rst:
```diff
@@ -0,0 +1,35 @@
+
+* **JaggedTensor**:
+    A JaggedTensor is a tensor with a jagged dimension which is dimension
+    whose slices may be of different lengths. Can optionally include weights
```
```suggestion
    its slices may be of different lengths. It can optionally include weights
```

svekars commented on docs/source/setup-torchrec.rst:
```diff
@@ -0,0 +1,136 @@
+
+Installations
+-------------
+Below we show installations for CUDA 12.1 as an example. For CPU or CUDA 11.8, swap "cu121" for "cpu" or "cu118".
```
We could also put these 4 options into tabs like this ([example result)](https://sphinx-design.readthedocs.io/en/latest/tabs.html#synchronised-tabs):
```
.. tab-set::

    .. tab-item:: Stable (pytorch.org)

        .. code-block:: bash
           pip install torch --index-url https://download.pytorch.org/whl/cu121
           pip install fbgemm-gpu --index-url https://download.pytorch.org/whl/cu121
           pip install torchmetrics==1.0.3
           pip install torchrec --index-url https://download.pytorch.org/whl/cu121

    .. tab-item:: Stable (PyPI, only CUDA 12.1)

        .. code-block:: bash
   	   pip install torch
   	   pip install fbgemm-gpu
    	   pip install torchrec
 ```

PaulZhang12 committed:
Set Up TorchRec Docs (#2434)

Summary:
Pull Request resolved: https://github.com/pytorch/torchrec/pull/2434

Documentation page for instructions on setting up TorchRec

Reviewed By: svekars

Differential Revision: D63548309
